src/Provider/Asset/AssetValueBuilderProvider.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Synolia\SyliusAkeneoPlugin\Provider\Asset;
+
+use Psr\Log\LoggerInterface;
+use Synolia\SyliusAkeneoPlugin\Builder\Asset\Attribute\AssetAttributeValueBuilderInterface;
+use Synolia\SyliusAkeneoPlugin\Exceptions\UnsupportedAttributeTypeException;
+
+final class AssetValueBuilderProvider implements AssetValueBuilderProviderInterface
+{
+    /** @var array<AssetAttributeValueBuilderInterface> */
+    private array $assetAttributeValueBuilders;
+
+    private LoggerInterface $akeneoLogger;
+
+    public function __construct(LoggerInterface $akeneoLogger)
+    {
+        $this->akeneoLogger = $akeneoLogger;
+    }
+
+    public function addBuilder(AssetAttributeValueBuilderInterface $assetAttributeValueBuilder): void
+    {
+        $this->assetAttributeValueBuilders[\get_class($assetAttributeValueBuilder)] = $assetAttributeValueBuilder;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return mixed|null
+     */
+    public function build(string $assetFamilyCode, string $assetCode, ?string $locale, ?string $scope, $value)
+    {
+        foreach ($this->assetAttributeValueBuilders as $assetAttributeValueBuilder) {
+            if ($assetAttributeValueBuilder->support($assetFamilyCode, $assetCode)) {
+                return $assetAttributeValueBuilder->build($assetFamilyCode, $assetCode, $locale, $scope, $value);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @return mixed|null
```

Comment:
we don't have BuilderInterface at least ?

Reply:
no, this is the "build" result which can be from various type. Here we are not returning the Provider object.